### PR TITLE
Separate concerns of creating, signing, proposing blocks

### PR DESCRIPTION
**Pull request by birchmd**
Separate processes of creating, signing and proposing new blocks; generate new genesis validator keys when node boots; allow specifying bond amounts and keys for genesis validators

## Overview
-Switch hash function to Blake2b256 for consistency with RSpace
-Casper is responsible for creating blocks from deployments
-The client is responsible for signing the created block to make it valid before proposing to the network
-Facility to obtain block created by Casper is exposed via gRPC to allow the use to handle signing however they like
-For convenience a default signing facility is supplied via the rnode command line
-Validators and bonds in genesis block can be specified in a file when the node boots up
-If validators are not specified then they are generated with a range of bonds

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/RHOL-341

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

**Comment by pyrocto** [comment]
When you say "The client is responsible for signing the created block", are you envisioning a situation where the one proposing the block to the network is not a validator?

**Comment by birchmd** [comment]
No. By "client" I mean client of the node software, who is a validator if they are proposing blocks. I'm saying that it is not necessarily the responsibility of the node software to sign blocks, validators are free to sign them by whatever pipeline they prefer. This is to address what came up in a meeting a couple weeks ago about possibly having multiple nodes sharing a common key, but being allowed to keep that key in a single location.

File: node/src/main/scala/coop/rchain/node/node.scala [hunk 1]
```diff
@@ -62,11 +68,45 @@ class NodeRuntime(conf: Conf)(implicit scheduler: Scheduler) {
...
   implicit val transportLayerEffect: TransportLayer[Task] = effects.transportLayer[Task](net)
 
+  val bondsFile: Option[File] =
+    conf.bondsFile.toOption
+      .flatMap(path => {
+        val f = new File(path)
+        if (f.exists) Some(f)
+        else {
+          Log[Task].warn(
+            s"Specified bonds file $path does not exist. Falling back on generating random validators."
```
Comment (pyrocto) [inline node/src/main/scala/coop/rchain/node/node.scala]: Is generating random validators just for testing?

**Comment by pyrocto** [comment]
Please resolve conflicts.

File: node/src/main/scala/coop/rchain/node/node.scala (hunk 1 above)
Comment (birchmd) [reply node/src/main/scala/coop/rchain/node/node.scala]: Yes.

**Pull request merged**